// mailosaur/cjs/mailosaur.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const models = __importStar(require("./models/index.js"));
const analysis_js_1 = __importDefault(require("./operations/analysis.js"));
const devices_js_1 = __importDefault(require("./operations/devices.js"));
const files_js_1 = __importDefault(require("./operations/files.js"));
const messages_js_1 = __importDefault(require("./operations/messages.js"));
const previews_js_1 = __importDefault(require("./operations/previews.js"));
const servers_js_1 = __importDefault(require("./operations/servers.js"));
const usage_js_1 = __importDefault(require("./operations/usage.js"));
const request_js_1 = __importDefault(require("./request.js"));
const mailosaurError_js_1 = __importDefault(require("./models/mailosaurError.js"));
/**
 * The Mailosaur client — the main entry point to the Mailosaur API. Construct an instance with
 * your API key (or set the `MAILOSAUR_API_KEY` environment variable), then use the operations
 * namespaces (`messages`, `servers`, `files`, `devices`, `analysis`, `previews`, `usage`) to
 * automate email and SMS testing.
 */
class MailosaurClient {
    /**
     * Returns an instance of the Mailosaur client.
     * @param apiKey Optional API key. Overrides the MAILOSAUR_API_KEY environment variable if set.
     * @param baseUrl Optionally overrides the base URL of the Mailosaur service.
     */
    constructor(apiKey, baseUrl) {
        const resolvedApiKey = apiKey || process.env.MAILOSAUR_API_KEY;
        if (!resolvedApiKey) {
            throw new Error("'apiKey' must be set via the MAILOSAUR_API_KEY environment variable, or passed to the MailosaurClient constructor.");
        }
        this.request = new request_js_1.default({
            baseUrl: baseUrl || 'https://mailosaur.com/',
            apiKey: resolvedApiKey,
        });
        this.analysis = new analysis_js_1.default(this);
        this.files = new files_js_1.default(this);
        this.messages = new messages_js_1.default(this);
        this.servers = new servers_js_1.default(this);
        this.usage = new usage_js_1.default(this);
        this.devices = new devices_js_1.default(this);
        this.previews = new previews_js_1.default(this);
        this.models = models;
    }
    httpError(response) {
        const httpStatusCode = response.statusCode;
        const httpResponseBody = response.body
            ? JSON.stringify(response.body)
            : null;
        let message = '';
        switch (httpStatusCode) {
            case 400:
                try {
                    if (!httpResponseBody) {
                        throw new Error('Empty response body');
                    }
                    const json = JSON.parse(httpResponseBody);
                    json.errors.forEach(err => {
                        message += `(${err.field}) ${err.detail[0].description}\r\n`;
                    });
                }
                catch (_e) {
                    message = 'Request had one or more invalid parameters.';
                }
                return new mailosaurError_js_1.default(message, 'invalid_request', httpStatusCode, httpResponseBody);
            case 401:
                return new mailosaurError_js_1.default('Authentication failed, check your API key.', 'authentication_error', httpStatusCode, httpResponseBody);
            case 403:
                return new mailosaurError_js_1.default('Insufficient permission to perform that task.', 'permission_error', httpStatusCode, httpResponseBody);
            case 404:
                return new mailosaurError_js_1.default('Not found, check input parameters.', 'invalid_request', httpStatusCode, httpResponseBody);
            case 410:
                return new mailosaurError_js_1.default('Permanently expired or deleted.', 'gone', httpStatusCode, httpResponseBody);
            default:
                return new mailosaurError_js_1.default('An API error occurred, see httpResponse for further information.', 'api_error', httpStatusCode, httpResponseBody);
        }
    }
}
exports.default = MailosaurClient;
__exportStar(require("./models/index.js"), exports);
// CommonJS compatibility - allows `const MailosaurClient = require('mailosaur')`
try {
    if (typeof module !== 'undefined' && module.exports) {
        module.exports = MailosaurClient;
        module.exports.default = MailosaurClient;
    }
}
catch (_a) {
    // Running in a true ESM context (e.g. vitest) where module.exports is read-only
}
