// mailosaur/esm/models/message.js
import MessageAddress from './messageAddress.js';
import MessageContent from './messageContent.js';
import Attachment from './attachment.js';
import Metadata from './metadata.js';
/**
 * The email or SMS message processed by Mailosaur.
 */
class Message {
    constructor(data = {}) {
        this.id = data.id;
        this.type = data.type;
        this.from = (data.from || []).map((i) => new MessageAddress(i));
        this.to = (data.to || []).map((i) => new MessageAddress(i));
        this.cc = (data.cc || []).map((i) => new MessageAddress(i));
        this.bcc = (data.bcc || []).map((i) => new MessageAddress(i));
        this.received = new Date(data.received);
        this.subject = data.subject;
        this.html = new MessageContent(data.html);
        this.text = new MessageContent(data.text);
        this.attachments = (data.attachments || []).map((i) => new Attachment(i));
        this.metadata = new Metadata(data.metadata);
        this.server = data.server;
    }
}
export default Message;
