// mailosaur/esm/mailosaur.js
import * as models from './models/index.js';
import Analysis from './operations/analysis.js';
import Devices from './operations/devices.js';
import Files from './operations/files.js';
import Messages from './operations/messages.js';
import Previews from './operations/previews.js';
import Servers from './operations/servers.js';
import Usage from './operations/usage.js';
import Request from './request.js';
import MailosaurError from './models/mailosaurError.js';
/**
 * The Mailosaur client — the main entry point to the Mailosaur API. Construct an instance with
 * your API key (or set the `MAILOSAUR_API_KEY` environment variable), then use the operations
 * namespaces (`messages`, `servers`, `files`, `devices`, `analysis`, `previews`, `usage`) to
 * automate email and SMS testing.
 */
class MailosaurClient {
    /**
     * Returns an instance of the Mailosaur client.
     * @param apiKey Optional API key. Overrides the MAILOSAUR_API_KEY environment variable if set.
     * @param baseUrl Optionally overrides the base URL of the Mailosaur service.
     */
    constructor(apiKey, baseUrl) {
        const resolvedApiKey = apiKey || process.env.MAILOSAUR_API_KEY;
        if (!resolvedApiKey) {
            throw new Error("'apiKey' must be set via the MAILOSAUR_API_KEY environment variable, or passed to the MailosaurClient constructor.");
        }
        this.request = new Request({
            baseUrl: baseUrl || 'https://mailosaur.com/',
            apiKey: resolvedApiKey,
        });
        this.analysis = new Analysis(this);
        this.files = new Files(this);
        this.messages = new Messages(this);
        this.servers = new Servers(this);
        this.usage = new Usage(this);
        this.devices = new Devices(this);
        this.previews = new Previews(this);
        this.models = models;
    }
    httpError(response) {
        const httpStatusCode = response.statusCode;
        const httpResponseBody = response.body
            ? JSON.stringify(response.body)
            : null;
        let message = '';
        switch (httpStatusCode) {
            case 400:
                try {
                    if (!httpResponseBody) {
                        throw new Error('Empty response body');
                    }
                    const json = JSON.parse(httpResponseBody);
                    json.errors.forEach(err => {
                        message += `(${err.field}) ${err.detail[0].description}\r\n`;
                    });
                }
                catch (_e) {
                    message = 'Request had one or more invalid parameters.';
                }
                return new MailosaurError(message, 'invalid_request', httpStatusCode, httpResponseBody);
            case 401:
                return new MailosaurError('Authentication failed, check your API key.', 'authentication_error', httpStatusCode, httpResponseBody);
            case 403:
                return new MailosaurError('Insufficient permission to perform that task.', 'permission_error', httpStatusCode, httpResponseBody);
            case 404:
                return new MailosaurError('Not found, check input parameters.', 'invalid_request', httpStatusCode, httpResponseBody);
            case 410:
                return new MailosaurError('Permanently expired or deleted.', 'gone', httpStatusCode, httpResponseBody);
            default:
                return new MailosaurError('An API error occurred, see httpResponse for further information.', 'api_error', httpStatusCode, httpResponseBody);
        }
    }
}
export default MailosaurClient;
export * from './models/index.js';
// CommonJS compatibility - allows `const MailosaurClient = require('mailosaur')`
try {
    if (typeof module !== 'undefined' && module.exports) {
        module.exports = MailosaurClient;
        module.exports.default = MailosaurClient;
    }
}
catch (_a) {
    // Running in a true ESM context (e.g. vitest) where module.exports is read-only
}
