// mailosaur/esm/request.js
import * as https from 'https';
import { HttpsProxyAgent } from 'https-proxy-agent';
import * as url from 'url';
const VERSION = '11.1.1';
class Request {
    constructor(options) {
        this.baseUrl = options.baseUrl || 'https://mailosaur.com/';
        const encodedKey = Buffer.from(`${options.apiKey}:`).toString('base64');
        this.headers = {
            Accept: 'application/json',
            Authorization: `Basic ${encodedKey}`,
            'User-Agent': options.agent || `mailosaur-node/${VERSION}`,
        };
        const httpProxy = process.env.https_proxy ||
            process.env.HTTPS_PROXY ||
            process.env.http_proxy ||
            process.env.HTTP_PROXY;
        if (httpProxy) {
            this.proxyAgent = new HttpsProxyAgent(httpProxy);
        }
    }
    buildOptions(method, path, options = {}) {
        const result = {
            method,
            url: `${this.baseUrl}${path}`,
            headers: {
                Accept: this.headers.Accept,
                Authorization: this.headers.Authorization,
                'User-Agent': this.headers['User-Agent'],
            },
        };
        if (options.buffer) {
            result.buffer = true;
        }
        if (options.qs) {
            const qs = options.qs;
            Object.keys(qs).forEach((key, index) => {
                let value = qs[key];
                if (!value) {
                    return;
                }
                if (value instanceof Date) {
                    value = value.toISOString();
                }
                const prefix = !index ? '?' : '&';
                result.url += `${prefix}${key}=${value}`;
            });
        }
        if (options.body) {
            const data = JSON.stringify(options.body);
            result.data = Buffer.from(data);
            result.headers['Content-Type'] = 'application/json';
            result.headers['Content-Length'] = String(Buffer.byteLength(result.data));
        }
        if (this.proxyAgent) {
            result.agent = this.proxyAgent;
        }
        return result;
    }
    invoke(options, callback) {
        let urlResult;
        try {
            urlResult = new url.URL(options.url);
        }
        catch (_ex) {
            urlResult = url.parse(options.url);
        }
        const spread = Object.assign(Object.assign({}, options), { protocol: urlResult.protocol || undefined, hostname: urlResult.hostname || undefined, path: (urlResult.pathname || '') + (urlResult.search || '') });
        const req = https.request(spread, res => {
            const data = [];
            res.on('data', (chunk) => data.push(chunk));
            res.on('end', () => {
                try {
                    if (data) {
                        res.body = options.buffer
                            ? Buffer.concat(data)
                            : JSON.parse(Buffer.concat(data).toString());
                    }
                }
                catch (_ex) {
                    res.body = data;
                }
                callback(null, res, res.body);
            });
        });
        req.on('error', (e) => callback(e));
        if (options.data) {
            req.write(options.data);
        }
        req.end();
    }
    request(method, path, options, callback) {
        let cb = callback;
        let opts = options;
        if (typeof options === 'function') {
            cb = options;
            opts = undefined;
        }
        const requestOptions = this.buildOptions(method, path, opts);
        if (!cb) {
            throw new Error('Callback is required.');
        }
        this.invoke(requestOptions, cb);
    }
    get(path, options, callback) {
        this.request('GET', path, options, callback);
    }
    put(path, options, callback) {
        this.request('PUT', path, options, callback);
    }
    post(path, options, callback) {
        this.request('POST', path, options, callback);
    }
    del(path, options, callback) {
        this.request('DELETE', path, options, callback);
    }
}
export default Request;
