// mailosaur/cjs/operations/servers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const serverListResult_js_1 = __importDefault(require("../models/serverListResult.js"));
const server_js_1 = __importDefault(require("../models/server.js"));
/**
 * Operations for creating and managing your Mailosaur inboxes (servers) — they
 * group your tests together, each with its own domain and SMTP/POP3/IMAP credentials.
 * Accessed via `client.servers`.
 */
class Servers {
    constructor(client) {
        this.client = client;
    }
    /**
     * Returns a list of your inboxes (servers). Inboxes (servers) are returned sorted in alphabetical order.
     * @returns A promise resolving to a {@link ServerListResult} containing your inboxes (servers).
     */
    async list() {
        const url = `api/servers`;
        return new Promise((resolve, reject) => {
            this.client.request.get(url, {}, (err, response, body) => {
                if (err) {
                    return reject(err);
                }
                if (!response || response.statusCode !== 200) {
                    return reject(response
                        ? this.client.httpError(response)
                        : new Error('No response received'));
                }
                resolve(new serverListResult_js_1.default(body));
            });
        });
    }
    /**
     * Creates a new inbox (server).
     * @param options Options used to create a new Mailosaur inbox (server).
     * @returns A promise resolving to the newly-created {@link Server}.
     */
    async create(options) {
        const url = `api/servers`;
        return new Promise((resolve, reject) => {
            this.client.request.post(url, { body: options }, (err, response, body) => {
                if (err) {
                    return reject(err);
                }
                if (!response || response.statusCode !== 200) {
                    return reject(response
                        ? this.client.httpError(response)
                        : new Error('No response received'));
                }
                resolve(new server_js_1.default(body));
            });
        });
    }
    /**
     * Retrieves the detail for a single inbox (server).
     * @param serverId The unique identifier of the inbox (server).
     * @returns A promise resolving to the {@link Server}.
     */
    async get(serverId) {
        const url = `api/servers/${serverId}`;
        return new Promise((resolve, reject) => {
            this.client.request.get(url, {}, (err, response, body) => {
                if (err) {
                    return reject(err);
                }
                if (!response || response.statusCode !== 200) {
                    return reject(response
                        ? this.client.httpError(response)
                        : new Error('No response received'));
                }
                resolve(new server_js_1.default(body));
            });
        });
    }
    /**
     * Retrieves the password for an inbox (server). This password can be used for SMTP, POP3, and IMAP connectivity.
     * @param serverId The unique identifier of the inbox (server).
     * @returns A promise resolving to the password for the inbox (server).
     */
    async getPassword(serverId) {
        const url = `api/servers/${serverId}/password`;
        return new Promise((resolve, reject) => {
            this.client.request.get(url, {}, (err, response, body) => {
                if (err) {
                    return reject(err);
                }
                if (!response || response.statusCode !== 200) {
                    return reject(response
                        ? this.client.httpError(response)
                        : new Error('No response received'));
                }
                resolve(body.value);
            });
        });
    }
    /**
     * Updates the attributes of an inbox (server).
     * @param serverId The unique identifier of the inbox (server).
     * @param server The updated inbox (server).
     * @returns A promise resolving to the updated {@link Server}.
     */
    async update(serverId, server) {
        const url = `api/servers/${serverId}`;
        return new Promise((resolve, reject) => {
            this.client.request.put(url, { body: server }, (err, response, body) => {
                if (err) {
                    return reject(err);
                }
                if (!response || response.statusCode !== 200) {
                    return reject(response
                        ? this.client.httpError(response)
                        : new Error('No response received'));
                }
                resolve(new server_js_1.default(body));
            });
        });
    }
    /**
     * Permanently delete an inbox (server). This will also delete all messages, associated attachments, etc. within the inbox (server). This operation cannot be undone.
     * @param serverId The unique identifier of the inbox (server).
     * @returns A promise resolving once the inbox (server) has been deleted.
     */
    async del(serverId) {
        const url = `api/servers/${serverId}`;
        return new Promise((resolve, reject) => {
            this.client.request.del(url, {}, (err, response) => {
                if (err) {
                    return reject(err);
                }
                if (!response || response.statusCode !== 204) {
                    return reject(response
                        ? this.client.httpError(response)
                        : new Error('No response received'));
                }
                resolve();
            });
        });
    }
    /**
     * Generates a random email address by appending a random string in front of the
     * domain name of the inbox (server).
     * @param serverId The identifier of the inbox (server).
     * @returns A random email address ending in the domain of the inbox (server).
     */
    generateEmailAddress(serverId) {
        const host = process.env.MAILOSAUR_SMTP_HOST || 'mailosaur.net';
        const random = (Math.random() + 1).toString(36).substring(7);
        return `${random}@${serverId}.${host}`;
    }
}
exports.default = Servers;
