// mailosaur/cjs/models/deliverabilityReport.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const emailAuthenticationResult_js_1 = __importDefault(require("./emailAuthenticationResult.js"));
const blockListResult_js_1 = __importDefault(require("./blockListResult.js"));
const content_js_1 = __importDefault(require("./content.js"));
const dnsRecords_js_1 = __importDefault(require("./dnsRecords.js"));
const spamAssassinResult_js_1 = __importDefault(require("./spamAssassinResult.js"));
/**
 * The results of deliverability report performed by Mailosaur.
 */
class DeliverabilityReport {
    constructor(data = {}) {
        this.spf = new emailAuthenticationResult_js_1.default(data.spf);
        this.dkim = (data.dkim || []).map((i) => new emailAuthenticationResult_js_1.default(i));
        this.dmarc = new emailAuthenticationResult_js_1.default(data.dmarc);
        this.blockLists = (data.blockLists || []).map((i) => new blockListResult_js_1.default(i));
        this.content = new content_js_1.default(data.content);
        this.dnsRecords = new dnsRecords_js_1.default(data.dnsRecords);
        this.spamAssassin = new spamAssassinResult_js_1.default(data.spamAssassin);
    }
}
exports.default = DeliverabilityReport;
