// mailosaur/cjs/models/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageTransactionListResult = exports.UsageTransaction = exports.UsageAccountLimits = exports.UsageAccountLimit = exports.SpamFilterResults = exports.SpamAssassinRule = exports.SpamAssassinResult = exports.SpamAnalysisResult = exports.ServerListResult = exports.ServerCreateOptions = exports.Server = exports.SearchOptions = exports.SearchCriteria = exports.PreviewRequestOptions = exports.PreviewListResult = exports.Preview = exports.OtpResult = exports.Metadata = exports.MessageSummary = exports.MessageReplyOptions = exports.MessageListResult = exports.MessageListOptions = exports.MessageHeader = exports.MessageForwardOptions = exports.MessageCreateOptions = exports.MessageContent = exports.MessageAddress = exports.Message = exports.MailosaurError = exports.Link = exports.Image = exports.EmailClientListResult = exports.EmailClient = exports.EmailAuthenticationResult = exports.DnsRecords = exports.DeviceListResult = exports.DeviceCreateOptions = exports.Device = exports.DeliverabilityReport = exports.Content = exports.Code = exports.BlockListResult = exports.Attachment = void 0;
const attachment_js_1 = __importDefault(require("./attachment.js"));
exports.Attachment = attachment_js_1.default;
const blockListResult_js_1 = __importDefault(require("./blockListResult.js"));
exports.BlockListResult = blockListResult_js_1.default;
const code_js_1 = __importDefault(require("./code.js"));
exports.Code = code_js_1.default;
const content_js_1 = __importDefault(require("./content.js"));
exports.Content = content_js_1.default;
const deliverabilityReport_js_1 = __importDefault(require("./deliverabilityReport.js"));
exports.DeliverabilityReport = deliverabilityReport_js_1.default;
const device_js_1 = __importDefault(require("./device.js"));
exports.Device = device_js_1.default;
const deviceCreateOptions_js_1 = __importDefault(require("./deviceCreateOptions.js"));
exports.DeviceCreateOptions = deviceCreateOptions_js_1.default;
const deviceListResult_js_1 = __importDefault(require("./deviceListResult.js"));
exports.DeviceListResult = deviceListResult_js_1.default;
const dnsRecords_js_1 = __importDefault(require("./dnsRecords.js"));
exports.DnsRecords = dnsRecords_js_1.default;
const emailAuthenticationResult_js_1 = __importDefault(require("./emailAuthenticationResult.js"));
exports.EmailAuthenticationResult = emailAuthenticationResult_js_1.default;
const emailClient_js_1 = __importDefault(require("./emailClient.js"));
exports.EmailClient = emailClient_js_1.default;
const emailClientListResult_js_1 = __importDefault(require("./emailClientListResult.js"));
exports.EmailClientListResult = emailClientListResult_js_1.default;
const image_js_1 = __importDefault(require("./image.js"));
exports.Image = image_js_1.default;
const link_js_1 = __importDefault(require("./link.js"));
exports.Link = link_js_1.default;
const mailosaurError_js_1 = __importDefault(require("./mailosaurError.js"));
exports.MailosaurError = mailosaurError_js_1.default;
const message_js_1 = __importDefault(require("./message.js"));
exports.Message = message_js_1.default;
const messageAddress_js_1 = __importDefault(require("./messageAddress.js"));
exports.MessageAddress = messageAddress_js_1.default;
const messageContent_js_1 = __importDefault(require("./messageContent.js"));
exports.MessageContent = messageContent_js_1.default;
const messageCreateOptions_js_1 = __importDefault(require("./messageCreateOptions.js"));
exports.MessageCreateOptions = messageCreateOptions_js_1.default;
const messageForwardOptions_js_1 = __importDefault(require("./messageForwardOptions.js"));
exports.MessageForwardOptions = messageForwardOptions_js_1.default;
const messageHeader_js_1 = __importDefault(require("./messageHeader.js"));
exports.MessageHeader = messageHeader_js_1.default;
const messageListOptions_js_1 = __importDefault(require("./messageListOptions.js"));
exports.MessageListOptions = messageListOptions_js_1.default;
const messageListResult_js_1 = __importDefault(require("./messageListResult.js"));
exports.MessageListResult = messageListResult_js_1.default;
const messageReplyOptions_js_1 = __importDefault(require("./messageReplyOptions.js"));
exports.MessageReplyOptions = messageReplyOptions_js_1.default;
const messageSummary_js_1 = __importDefault(require("./messageSummary.js"));
exports.MessageSummary = messageSummary_js_1.default;
const metadata_js_1 = __importDefault(require("./metadata.js"));
exports.Metadata = metadata_js_1.default;
const otpResult_js_1 = __importDefault(require("./otpResult.js"));
exports.OtpResult = otpResult_js_1.default;
const preview_js_1 = __importDefault(require("./preview.js"));
exports.Preview = preview_js_1.default;
const previewListResult_js_1 = __importDefault(require("./previewListResult.js"));
exports.PreviewListResult = previewListResult_js_1.default;
const previewRequestOptions_js_1 = __importDefault(require("./previewRequestOptions.js"));
exports.PreviewRequestOptions = previewRequestOptions_js_1.default;
const searchCriteria_js_1 = __importDefault(require("./searchCriteria.js"));
exports.SearchCriteria = searchCriteria_js_1.default;
const searchOptions_js_1 = __importDefault(require("./searchOptions.js"));
exports.SearchOptions = searchOptions_js_1.default;
const server_js_1 = __importDefault(require("./server.js"));
exports.Server = server_js_1.default;
const serverCreateOptions_js_1 = __importDefault(require("./serverCreateOptions.js"));
exports.ServerCreateOptions = serverCreateOptions_js_1.default;
const serverListResult_js_1 = __importDefault(require("./serverListResult.js"));
exports.ServerListResult = serverListResult_js_1.default;
const spamAnalysisResult_js_1 = __importDefault(require("./spamAnalysisResult.js"));
exports.SpamAnalysisResult = spamAnalysisResult_js_1.default;
const spamAssassinResult_js_1 = __importDefault(require("./spamAssassinResult.js"));
exports.SpamAssassinResult = spamAssassinResult_js_1.default;
const spamAssassinRule_js_1 = __importDefault(require("./spamAssassinRule.js"));
exports.SpamAssassinRule = spamAssassinRule_js_1.default;
const spamFilterResults_js_1 = __importDefault(require("./spamFilterResults.js"));
exports.SpamFilterResults = spamFilterResults_js_1.default;
const usageAccountLimit_js_1 = __importDefault(require("./usageAccountLimit.js"));
exports.UsageAccountLimit = usageAccountLimit_js_1.default;
const usageAccountLimits_js_1 = __importDefault(require("./usageAccountLimits.js"));
exports.UsageAccountLimits = usageAccountLimits_js_1.default;
const usageTransaction_js_1 = __importDefault(require("./usageTransaction.js"));
exports.UsageTransaction = usageTransaction_js_1.default;
const usageTransactionListResult_js_1 = __importDefault(require("./usageTransactionListResult.js"));
exports.UsageTransactionListResult = usageTransactionListResult_js_1.default;
