// mailosaur/cjs/models/message.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const messageAddress_js_1 = __importDefault(require("./messageAddress.js"));
const messageContent_js_1 = __importDefault(require("./messageContent.js"));
const attachment_js_1 = __importDefault(require("./attachment.js"));
const metadata_js_1 = __importDefault(require("./metadata.js"));
/**
 * The email or SMS message processed by Mailosaur.
 */
class Message {
    constructor(data = {}) {
        this.id = data.id;
        this.type = data.type;
        this.from = (data.from || []).map((i) => new messageAddress_js_1.default(i));
        this.to = (data.to || []).map((i) => new messageAddress_js_1.default(i));
        this.cc = (data.cc || []).map((i) => new messageAddress_js_1.default(i));
        this.bcc = (data.bcc || []).map((i) => new messageAddress_js_1.default(i));
        this.received = new Date(data.received);
        this.subject = data.subject;
        this.html = new messageContent_js_1.default(data.html);
        this.text = new messageContent_js_1.default(data.text);
        this.attachments = (data.attachments || []).map((i) => new attachment_js_1.default(i));
        this.metadata = new metadata_js_1.default(data.metadata);
        this.server = data.server;
    }
}
exports.default = Message;
