// mailosaur/cjs/models/messageSummary.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const messageAddress_js_1 = __importDefault(require("./messageAddress.js"));
/**
 * A summary of the message processed by Mailosaur. This summary does not include
 * the contents of the email or SMS message, for which you will need the full
 * message object.
 */
class MessageSummary {
    constructor(data = {}) {
        this.id = data.id;
        this.type = data.type;
        this.server = data.server;
        this.from = (data.from || []).map((i) => new messageAddress_js_1.default(i));
        this.to = (data.to || []).map((i) => new messageAddress_js_1.default(i));
        this.cc = (data.cc || []).map((i) => new messageAddress_js_1.default(i));
        this.bcc = (data.bcc || []).map((i) => new messageAddress_js_1.default(i));
        this.received = new Date(data.received);
        this.subject = data.subject;
        this.summary = data.summary;
        this.attachments = data.attachments || 0;
    }
}
exports.default = MessageSummary;
